apitools/base/py/credentials_lib.py

```diff
@@ -171,17 +182,15 @@ def __init__(self, scopes=None, service_account_name='default', **kwds):
         # identified these scopes in the same execution. However, the
         # available scopes don't change once an instance is created,
         # so there is no reason to perform more than one query.
-        #
-        # TODO(craigcitro): Move this into oauth2client.
         self.__service_account_name = service_account_name
-        cache_filename = None
         cached_scopes = None
-        if 'cache_filename' in kwds:
-            cache_filename = kwds['cache_filename']
+        cache_filename = kwds.get('cache_filename')
+        if cache_filename:
             cached_scopes = self._CheckCacheFileForMatch(
                 cache_filename, scopes)
 
-        scopes = cached_scopes or self._ScopesFromMetadataServer(scopes)
+        scopes = (scopes or cached_scopes or
```

Comment:
so note that before this change, *if* `cache_filename` was provided, and there were cached scopes, then we totally ignored the `scopes` argument provided by the user.

in particular, it means that it's possible the user would get back a token for different scopes than they requested.